zion/src/main/scala/edu/uci/ics/cloudberry/zion/model/impl/AsterixQueryGenerator.scala

```diff
@@ -0,0 +1,270 @@
+package edu.uci.ics.cloudberry.zion.model.impl
+
+import edu.uci.ics.cloudberry.zion.model.datastore.{IQLGenerator, QueryParsingException}
+import edu.uci.ics.cloudberry.zion.model.schema._
+
+
+/**
+  * Defines constant strings for query languages supported by AsterixDB
+  */
+trait TypeImpl {
+
+  val aggregateFuncMap: Map[AggregateFunc, String]
+
+  def getAggregateStr(aggregate: AggregateFunc): String = {
+    aggregateFuncMap.get(aggregate) match {
+      case Some(impl) => impl
+      case None => throw new QueryParsingException(s"No implementation is provided for aggregate function ${aggregate.name}")
+    }
+  }
+
+  val datetime: String = "datetime"
+  val round: String = "round"
+
+  val dayTimeDuration: String
+  val yearMonthDuration: String
+  val getIntervalStartDatetime: String
+  val intervalBin: String
+
+  val spatialIntersect: String
+  val createRectangle: String
+  val createPoint: String
+  val spatialCell: String
+  val getPoints: String
+
+
+  val similarityJaccard: String
+  val contains: String
+  val wordTokens: String
+
+}
+
+abstract class AsterixQueryGenerator extends IQLGenerator {
+
+  /**
+    * represent the expression for a [[Field]]
+    *
+    * @param refExpr the expression for referring this field by the subsequent statements
+    * @param defExpr the expression the defines the field
+    *
+    */
+  case class FieldExpr(refExpr: String, defExpr: String)
+
+  /**
+    * Partial parse results after parsing each [[Statement]]
+    *
+    * @param strs    a sequence of parsed query strings, which would be composed together later.
+    * @param exprMap a new field expression map
+    */
+  case class PartialResult(strs: Seq[String], exprMap: Map[String, FieldExpr])
+
+  protected def typeImpl: TypeImpl
+
+  protected def quote: String
+
+  protected def sourceVar: String
+
+  protected def lookupVar: String
+
+  protected def unnestVar: String
+
+  protected def groupVar: String
+
+  protected def globalAggrVar: String
+
+  protected def outerSelectVar: String
+
+  protected def suffix: String
```

Comment:
Some query language (like SQL++) requires ";" after each query, while AQL does not. Suffix actually represents this ";" appended to each query. Just added some doc to this.